scylla-cdc/src/lib.rs

```diff
@@ -5,9 +5,440 @@ pub mod stream_generations;
 
 #[cfg(test)]
 mod tests {
-    #[test]
-    fn it_works() {
-        let result = 2 + 2;
-        assert_eq!(result, 4);
+    use crate::consumer::*;
+    use anyhow::{anyhow, Result};
+    use async_trait::async_trait;
+    use futures::StreamExt;
+    use itertools::Itertools;
+    use scylla::frame::response::result::CqlValue;
+    use scylla::frame::value::{Value, ValueTooBig};
+    use scylla::prepared_statement::PreparedStatement;
+    use scylla::{Session, SessionBuilder};
+    use std::collections::{HashMap, VecDeque};
+    use std::hash::Hash;
+    use std::sync::Arc;
+    use tokio::sync::Mutex;
+
+    type OperationsMap = Arc<Mutex<HashMap<PrimaryKeyValue, VecDeque<Operation>>>>;
+
+    // The driver's CqlValue cannot be used as HashMap key,
+    // because it doesn't have the Eq trait.
+    #[derive(Debug, Eq, PartialEq, Hash)]
+    enum PrimaryKeyValue {
+        // Name consistency with CqlValue from the driver is recommended.
+        Int(i32),
+        Text(String),
+        List(Vec<PrimaryKeyValue>),
+        ComplexPK(Vec<PrimaryKeyValue>),
```

Comment:
Can't we just use `Vec<PrimaryKeyValue>` instead of `PrimaryKeyValue` as the key in `OperationsMap`? This should eliminate the need for `ComplexPK`.

Reply:
I thought it was more intuitive in the case of using a one-column pk. Maybe you're right, I'll have to see which one looks better.